cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/writer/MessagingLineageWriter.java

```diff
@@ -81,7 +83,16 @@ private void publishLineage(ProgramRunId programRunId, DataAccessLineage lineage
   }
 
   private void publish(MetadataMessage message) {
-    StoreRequest request = StoreRequestBuilder.of(topic).addPayload(GSON.toJson(message)).build();
+    String messageJson = GSON.toJson(message);
+    int lineageSize = messageJson.length();
+
+    if (lineageSize > publishSizeLimit) {
+      LOG.warn("Size of lineage message is {} bytes, which is larger than the limit {}.\n" +
+        "Therefore the lineage will not be published.", lineageSize, publishSizeLimit);
```

Comment:
Could you please just log this once as warn and later as trace? Otherwise we may flood log with warnings

Reply:
Done.